cvmfs/catalog.cc

```diff
@@ -63,6 +63,9 @@ Catalog::Catalog(const PathString &path,
   int retval = pthread_mutex_init(lock_, NULL);
   assert(retval == 0);
 
+  atomic_init32(&external_data_status_);
```

Comment:
This is always set in a locked context, isn't it?  So `external_data_status_` can just be of type `ExternalDataStatus`